### DISCUSSION TITLE
Creating your own project

### DISCUSSION THREAD

**Discussion by catherineisonline** [catherineisonline/pizza-time-with-react#6] [General]
> kingalfarizi: hey can u teach me how i can connect my database? im trying for two weeks now i cant figure it out :(
> 
> kingalfarizi: ive done following the documentation and yet i cant still use login/register, the loading is endless

@kingalfarizi Would be hard to help without seeing the entire code, the issues can be different. Do you have the code anywhere where I can see it by chance?

- Make sure that you created the environment variables 
- Make sure that you deployed your own backend (if you are using Vercel google how to create a Vercel backend)
- Make sure that you add environment variables to Vercel as well! (You need to go to the project's settings in Vercel and then on the left menu choose Variable Environments)
- Check what the console says when you try to log in and what error it gives you, if any?

**Comment by kingalfarizi** [comment]
i have all that finished, unfortunately i didnt get  any error in the console. can i have the backend with the database by any chance? thanks in advance catherine :) this is so important for my final project in university :D

**Comment by catherineisonline** [comment]
Unfortunately, I can't help without seeing the code and I cannot share my database because I have free limits that I am using for various projects. 
Instead of adding a backend via Vercel, did you try to run the server locally first and make sure it works at least locally?

**Comment by catherineisonline** [comment]
A quick checklist of what you can do to check it locally:
-Fork the project from a new branch which I created with your username
-Then create the file .env and add it where the package.json is located 
-Inside the .env create variables:

REACT_APP_DATABASE_URL=find this in the documentation
REACT_APP_USERS_URL=http://localhost:3000/users
MYSQL_ATTR_SSL_CA=find this in the documentation
REACT_APP_CAPTCHA_KEY=find this in the reCaptcha documentation
REACT_APP_CAPTCHA_SECRET=find this in the reCaptcha documentation

-Next install the node modules
-Next run the server with command "node backend/api/index.mjs" 
-Next open in the browser http://localhost:3000/ and you should see a white screen that says "Server Deployed 🥳"
-Got to http://localhost:3000/users and on another port open the website and register
-In the path http://localhost:3000/users after registration you should be able to see the the users but also make sure that you created a table in the database so it's like a shell to store the users. To create a table you need to open the https://app.planetscale.com/the project name/the database name which should be users in my case. Then you need to open the console and in the console you should create using SQL a table with the columns: id | email | password | fullname | address | number
You can google how to create tables, don't remember exactly.

So once everything is done, you will either get stuck at the very start where server doesn't deploy locally and that is one issue or you will get stuck when trying to get users which means there is issue with connecting to the database. If you try to register and database doesn't update it means you are sending the wrong data.

**Reply by kingalfarizi** [reply]
hey its me again :)
i've managed to do what you tell me...
but if i want to register, the database does not updated as you said....
it says "Seems like this email has already been registered!"
if i create the data manually, i can login
i checked all the variables and im sure all is right...
what should i double check?
Thanks catherine :D

**Reply by catherineisonline** [reply]
Hello. Well, you need to check the full data, what exactly are you sending? I don't remember the exact lines but there is the final object that will be sent to the database so could you check what is inside, is the structure exact? And if it says "Seems like this email has already been registered!" check the condition and when it says it, what is the condition it has to say that and check if you meet that condition by accident. Hard to say much without having the full understanding of your code!
